autotest/test_modeltime.py

```diff
@@ -0,0 +1,292 @@
+import datetime
+
+import numpy as np
+import pandas as pd
+
+import flopy
+from flopy.discretization.modeltime import ModelTime
+
+
+def test_date_userinput_parsing():
+    formats = [
+        datetime.datetime(2024, 11, 12),
+        np.datetime64("2024-11-12"),
+        pd.Timestamp("2024-11-12"),
+        "2024-11-12",
+        "2024/11/12",
+        "11-12-2024",
+        "11/12/2024",
+    ]
+
+    valid = datetime.datetime(2024, 11, 12)
```

Comment:
for simplicity and readability, this could be shortened to simply
```python
assert ModelTime.parse_datetime(dt_rep) == datetime.datetime(2024, 11, 12)
```
same for other checks with ``AssertionError``s